conda/environments/all_cuda-122.yaml

```diff
@@ -3,16 +3,17 @@
 channels:
 - legate
 - legate/label/experimental
+- legate/label/rc
 - conda-forge
 - nvidia
 dependencies:
 - cmake>=3.24.0,!=3.30.0
 - cuda-cudart-dev
 - cuda-nvcc
 - cuda-version>=12.2
-- cunumeric==25.01.*,>=0.0.0.dev0
+- cupynumeric==24.11.*
```

Comment:
@jameslamb may have an opinion on this.  But right now things are tightly pinned for legate, and we don't need the `>=0.0.0.dev0` part for a release.

So I think this is fine (we need to change it back to 25.01, basically immediately anyway).

Reply:
This is good. `legate-boost` directly imports `cupynumeric`, so it makes sense to me for it to maintain its own pins on that instead of only relying on getting it transitively from `legate`.

> *we don't need the `>=0.0.0.dev0` part for a release*

To clarify... in conda world, something like `24.11.*` can still match to versions like `24.11.00.dev8`, `24.11.00.a8`, or `24.11.00.rc1`.

```shell
conda create \
    --dry-run \
    --name delete-me \
    --override-channels \
    --channel legate \
    --channel legate/label/experimental \
    --channel conda-forge \
    --channel nvidia \
    'cupynumeric==25.01.*'
```

Solves like this:

```text
  cupynumeric        legate/label/experimental/linux-64::cupynumeric-25.01.00.dev25-cuda12_py311_gcba82286_25_cpu
  ...
  legate             legate/label/experimental/linux-64::legate-25.01.00.dev38-cuda12_py311_g838956e0_38_ucx_cpu
```

So removing it here doesn't offer any protection against picking up nightlies... the only way to get 100% protection against that with a pin like `24.11.*` is to keep the `legate/label/experimental` label out of the environment.

I'd only recommended having these extra specs like `,>=0.0.0.dev0` because it *does* matter for `pip`... it's a way to tell `pip` you're willing to accept pre-releases for a specific dependency, which is generally safer than allowing them for all dependencies (via `pip install --pre`). I know these libraries aren't currently publishing wheels, but they might in the future.

If it simplifies things, it'd be safe to not ever use `,>=0.0.0dev0` for as long as these `legate-*` projects are conda-only.

Reply:
@jameslamb will get your input before we do the 25.01 tag.